test/tpmtest/tpmtest.c

```diff
@@ -458,6 +460,95 @@ void GetTpmVersion()
     }
 }
 
+// Get the TPM 2.0 commands supported by the TPM.
+TSS2_RC GetCommands( TSS2_SYS_CONTEXT *resMgrSysContext, TPML_CCA **supportedCommands )
+{
+    UINT32 numCommands;
+    TPMI_YES_NO moreData;
+    TPMS_CAPABILITY_DATA capabilityData;
+    TPMA_CC *commandPtr;
+    TSS2_RC rval = TSS2_RC_SUCCESS;
+    UINT32 i;
+
+    // First get the number of commands
+    rval = Tss2_Sys_GetCapability( resMgrSysContext, 0,
+            TPM_CAP_TPM_PROPERTIES, TPM_PT_TOTAL_COMMANDS,
+            1, 0, &capabilityData, 0 );
+    if( rval == TPM_RC_SUCCESS &&
+            capabilityData.capability == TPM_CAP_TPM_PROPERTIES &&
+            capabilityData.data.tpmProperties.count == 1 &&
+            capabilityData.data.tpmProperties.tpmProperty[0].property == TPM_PT_TOTAL_COMMANDS )
+    {
+        numCommands = capabilityData.data.tpmProperties.tpmProperty[0].value;
+    }
+    else
+    {
+        goto returnFromGetCommands;
+    }
+
+    // Allocate memory for them
+    *supportedCommands = (TPML_CCA *)malloc( numCommands * sizeof( TPMA_CC ) + sizeof( UINT32 ) );
+    if( !*supportedCommands )
+    {
+        rval = TSS2_BASE_RC_INSUFFICIENT_BUFFER + TSS2_RESMGR_ERROR_LEVEL;
```

Comment:
I know that this is a rare condition but if a memory allocation in this test program fails the error that will be returned to the user will indicate that the error came from the resource manager. This is really misleading and could lead to some really painful (and useless) bug hunts in the resource manager. If the application fails to allocate memory there's little that the caller can do other than report the issue to the caller / user and exit. I'd think a simple `printf` with information about the failure would suffice. In this case the caller will probably just want to terminate the program.